File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.py
def solution(operations):
    answer = []
    heap = []

    for i in operations:
        if i == "D -1":
            if len(heap) != 0:
                heap.remove(min(heap))
        elif i == "D 1":                       
            if len(heap) != 0:
                heap.remove(max(heap))
        else:
            heap.append(int(i[2:len(i)]))

    if len(heap) > 0:
        answer.append(max(heap))
        answer.append(min(heap))
    else:
        answer.append(0)
        answer.append(0)
    return answer

File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.java
import java.util.*;

class Solution {
    public int[] solution(String[] operations) {
        int[] answer = new int[2];
        PriorityQueue<Integer> q = new PriorityQueue<>();
        
        for(int i = 0; i < operations.length; i++){
            String[] str = operations[i].split(" ");
            if (str[0].equals("I")) {
                q.add(Integer.parseInt(str[1]));
            } else if (!str[0].equals("I") && q.size() > 0) {
                if (str[1].contains("-")) {
                    q.remove(Collections.min(q)); 
                } else {
                    q.remove(Collections.max(q));
                }
            }
        }
        
        if (q.size() == 0) {  
            answer[0] = 0;
            answer[1] = 0;
        } else {
            answer[0] = Collections.max(q);
            answer[1] = Collections.min(q);
        }
        
        return answer;
    }
}